.github/workflows/ci.yml

```diff
@@ -68,3 +68,31 @@ jobs:
 
       - name: Run cargo tests
         run: cargo test --all-features
+
+  check:
+    if: always()
+    needs: [test, lint]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check if all needed jobs succeeded
+        uses: re-actors/alls-green@release/v1
+        with:
+          jobs: ${{ toJSON(needs) }}
+
+  release:
+    needs: [check]
+    if: "success() && startsWith(github.ref, 'refs/tags/')"
```

Comment:
Ensure the release only happens on a successful check and when the ref is a tag.